tests/XsdDateCodec.spec.js

```diff
@@ -0,0 +1,70 @@
+/*!
+* Copyright (c) 2020 Digital Bazaar, Inc. All rights reserved.
+*/
+import {
+  expect
+} from 'chai';
+import {XsdDateCodec} from '../lib/codecs/XsdDateCodec';
+import {MockEncoder} from './MockEncoder';
+
+describe('XsdDateCodec', () => {
+  it('should roundtrip encode-decode successfully', async () => {
+    const input = '2020-07-14';
+    const mockEncoder = new MockEncoder();
+
+    const encoderCodec = new XsdDateCodec();
+    encoderCodec.set({value: input});
+    encoderCodec.encodeCBOR(mockEncoder);
+
+    const decoderCodec = new XsdDateCodec();
+    decoderCodec.set({value: mockEncoder.result});
+    const decodedResult = decoderCodec.decodeCBOR();
+
+    expect(input).equal(decodedResult);
+  });
+
+  it('should throw exception when input includes time component', async () => {
+    const input = '2020-07-14T19:23:24Z';
+    const mockEncoder = new MockEncoder();
+
+    const encoderCodec = new XsdDateCodec();
+    encoderCodec.set({value: input});
+    expect(() => encoderCodec.encodeCBOR(mockEncoder))
+      .to
+      .throw('Invalid input, date includes time \
+component lost of precision expected');
```

Comment:
```suggestion
    expect(() => encoderCodec.encodeCBOR(mockEncoder))
      .to
      .throw('Invalid input, date includes time \
component lost of precision expected');
```

This sort of expect logic tends to be hard to read... we try to avoid wrapping statements in `expect(...)` and having multi-line expect statements. Fine for now, but needs to be updated to be easier to read.